src/Config.cpp

```diff
@@ -55,7 +55,9 @@ using nonstd::optional;
 #  define DLLIMPORT
 #endif
 
+#ifndef environ
```

Comment:
For my own understanding: I guess this means that when using Clang on Windows there is an `environ` preprocessor macro already defined? What makes this an issue when using Clang but not other compilers? What does the build error look like?

Reply:
Here: https://github.com/orgads/MINGW-packages/runs/3456109063?check_suite_focus=true
```
  FAILED: src/CMakeFiles/ccache_framework.dir/Config.cpp.obj 
  D:\a\_temp\msys\msys64\clang64\bin\clang++.exe -DHAVE_REDIS_STORAGE_BACKEND -DUSE_XXH_DISPATCH -DWIN32_LEAN_AND_MEAN -DXXH_STATIC_LINKING_ONLY -Dnssv_CONFIG_SELECT_STRING_VIEW=nssv_STRING_VIEW_NONSTD -IC:/_/mingw-w64-ccache/src/build-x86_64-w64-mingw32 -IC:/_/mingw-w64-ccache/src/ccache-4.4/src -march=x86-64 -mtune=generic -O2 -pipe -O3 -DNDEBUG -include C:/_/mingw-w64-ccache/src/build-x86_64-w64-mingw32/config.h -Wall -Wextra -Wnon-virtual-dtor -Wcast-align -Wunused -Woverloaded-virtual -Wpedantic -Wdouble-promotion -Werror -Wno-shadow-field-in-constructor -Wno-c++2a-compat -Wno-zero-as-null-pointer-constant -Wno-undefined-func-template -pthread -D_FILE_OFFSET_BITS=64 -std=c++14 -MD -MT src/CMakeFiles/ccache_framework.dir/Config.cpp.obj -MF src\CMakeFiles\ccache_framework.dir\Config.cpp.obj.d -o src/CMakeFiles/ccache_framework.dir/Config.cpp.obj -c C:/_/mingw-w64-ccache/src/ccache-4.4/src/Config.cpp
  C:/_/mingw-w64-ccache/src/ccache-4.4/src/Config.cpp:58:25: error: '__p__environ' redeclared without 'dllimport' attribute: previous 'dllimport' ignored [-Werror,-Winconsistent-dllimport]
  DLLIMPORT extern char** environ;
                          ^
  D:/a/_temp/msys/msys64/clang64/x86_64-w64-mingw32/include/stdlib.h:704:17: note: expanded from macro 'environ'
  #define environ _environ
                  ^
  D:/a/_temp/msys/msys64/clang64/x86_64-w64-mingw32/include/stdlib.h:263:21: note: expanded from macro '_environ'
  #define _environ (* __p__environ())
                      ^
  D:/a/_temp/msys/msys64/clang64/x86_64-w64-mingw32/include/stdlib.h:185:27: note: previous declaration is here
    _CRTIMP char ***__cdecl __p__environ(void);
                            ^
  D:/a/_temp/msys/msys64/clang64/x86_64-w64-mingw32/include/stdlib.h:185:3: note: previous attribute is here
    _CRTIMP char ***__cdecl __p__environ(void);
    ^
  D:/a/_temp/msys/msys64/clang64/x86_64-w64-mingw32/include/_mingw.h:59:40: note: expanded from macro '_CRTIMP'
  #      define _CRTIMP  __attribute__ ((__dllimport__))
                                         ^
  1 error generated.
```

Reply:
Thanks.